Expose ProcedureGuard to check tx termination and guards from within …

boggle opened the pull request:
…long running procedures

boggle committed:
Expose ProcedureGuard to check tx termination and guards from within long running procedures

boggle committed:
Correct default time unit that is being used by Settings

boggle committed:
Rename ProcedureGuard to TerminationGuard and move to procedure api package

MishaDemianenko commented on community/kernel/src/main/java/org/neo4j/kernel/guard/TimeoutGuard.java:
```diff
@@ -48,7 +48,7 @@ public void check( KernelStatement statement )
     }
 
-    private void check( KernelTransactionImplementation transaction )
+    public void check( KernelTransactionImplementation transaction )
```
Please add `@Override` here

MishaDemianenko commented on community/kernel/src/main/java/org/neo4j/kernel/impl/factory/DataSourceModule.java:
```diff
@@ -421,6 +428,37 @@ private Procedures setupProcedures( PlatformModule platform, EditionModule editi
+        return (ctx) ->
+        {
+            KernelTransaction ktx = ctx.get( KERNEL_TRANSACTION );
+            return (TerminationGuard) () ->
```
please extract this class into separate class and put it into some procedure related package, most probably where `TerminationGuard` live

MishaDemianenko reviewed (changes requested):
@boggle  Two small comments and also please note there is test failure atm. Review round complete

boggle committed:
Address feedback from code review

henriknyman committed:
Fix race condition in test

The shouldTerminateLongRunningProcedureThatChecksTheGuardRegularlyIfKilled test
could try to extract query id and kill the query before the actual query execution started.

boggle committed:
Fix use of latches in NeverEndingProcedure

boggle commented:
This was green; just rebased but since it's weekend waiting for another build.
